pkg/logql/engine.go

```diff
@@ -140,23 +140,29 @@ func (opts *EngineOpts) applyDefault() {
 	}
 }
 
-// Engine is the LogQL engine.
-type Engine struct {
+type Engine interface {
```

Comment:
Would it not be simpler to name this something else, to avoid all the refactoring?
`NewDefaultEngine` doesn't really mean anything.

Reply:
Fair point but what would you name it? What do you think about `QueryFactory`?

Reply:
It's not a factory, though...
Idiomatic Go would suggest `Querier`, which seems kinda reasonable here?

Reply:
It is a factory. The method `Query()` constructs a `Query` that can be executed via `Exec()`. It is a little confusing.

Reply:
What I meant is: the interface is not a factory.
Anyway I don't mind what it's called; but let's rather rename this than change the engine name everywhere I think

Reply:
I beg to differ (https://en.wikipedia.org/wiki/Factory_(object-oriented_programming)). Although, I'm wondering if the implementations are factories and this is a factory interface (https://www.baeldung.com/java-abstract-factory-pattern). Java...

Anyhow, I'll come up with something else.